src/components/UI/FormElement/FormElement.js

```diff
@@ -42,6 +44,30 @@ const FormElement = (props) => {
         </ul>
       );
       break;
+    case ('file'):
```

Comment:
In my opinion, each of the cases within this switch should be their own React components sitting in their own files. They should each receive the entire `props` spread onto themselves (e.g. `<TextFormElement {...props} />`) and have all associated logic handled inside the specific element, rather than in this switch.

More generally, a function should do one thing. If a function is orchestrating (e.g. this switch, which is _totally cool_), that is one thing. If somehow the orchestrator becomes a facilitator or an actual builder or factory or something else, you've got too many contexts to keep track of at one level, leading to confusion, bad reasoning, and needless bugs.

Reply:
@shoelessness Yep, makes sense. I doubt this component will ever be anything more than an orchestrator though, so do you think I should split it up or was this more of just a general comment?

Reply:
So, what you have here isn't just an orchestrator. An orchestrator would check for different cases and then delegate the necessary operations to other methods. This orchestrates and also does the work.

A simple orchestrator would look more like:

```
const FormElement = (props) => {
switch(props.elementType) {
case ('input'):
return <InputFormElement {...props} />;
...
```

Where `InputFormElement` would be in its own file and contain everything else within that switch case.

Reply:
@shoelessness Ok, so how would I loop through the formElementsArray and select the right component? There won't just be one so wouldn't I need to do the logic to check multiple times? Since there's multiple types of form elements in one form.